packages/components/pie-webc-core/src/mixins/rtl/rtlMixin.ts

```diff
@@ -21,11 +21,23 @@ declare class _RTLInterface {
     isRTL: boolean;
 }
 
+/**
+ * This RTL mixin is used with Lit Web components to add programmatic Right-to-Left (RTL) support.
+ * It is only required if your component either:
+ * - Needs RTL awareness in its TypeScript logic.
+ * - Its CSS requires a [dir='rtl'] attribute to be present.
+ *
+ * By default, the 'dir' property will be inferred from a 'dir' attribute set in the document root.
+ * It can be manually set if needed, but by default, it's not required. The 'dir' property value is
+ * reflected in the DOM, making it queryable. The 'isRTL' internal property returns true or false
+ * depending on whether or not the 'dir' property is 'rtl' or not.
+ */
+
 export const RtlMixin =
     <T extends Constructor<LitElement>>(superClass: T) => {
         class RTLElement extends superClass implements _RTLInterface {
             @property({ type: String, reflect: true })
-                dir : htmlDirAttribute = 'ltr';
```

Comment:
@raoufswe we may want to test for the existence of the document and documentElement. Otherwise, this could possibly cause build issues for consumers during SSR

Reply:
Fixed